### ISSUE TITLE
different line height / vertical spacing / between urlwatch 2.22 and 2.28

### ISSUE THREAD

**Issue by kongomongo** [thp/urlwatch#774]
Hi there,

I noticed the following after upgrading debian buster to bullseye. I upgraded from urlwatch 2.22 to 2.28 in the process, did not change anything else and the following changed in my HTML output (rendered by Outlook):

2.22:
![image](https://github.com/thp/urlwatch/assets/5646390/b486b0e8-7655-48d4-a2d3-d9187126ec40)

2.28:
![image](https://github.com/thp/urlwatch/assets/5646390/f7eea685-5aa1-4481-8d4c-b85da825c3b5)

Is this on purpose? Is this something I can change? 

This is the HTML diff:

![image](https://github.com/thp/urlwatch/assets/5646390/0c1f51fd-2a21-4ee5-87f3-9e083c7e579f)

It seems this CSS is hardcoded, any way to change it via YAML?

Many thanks!

**Comment by thp** [comment]
Seems like the first part at least is https://github.com/thp/urlwatch/pull/730

**Comment by kongomongo** [comment]
Yes the coloring was changed in #730 but also the line-height: 1.5 was added which I am really not a fan of. I like my list of changes compact.

What do you suggest? Expose that as a variable? Make a "compact" setting?

**Comment by kongomongo** [comment]
Sorry, found another difference: When using iOS (dark mode enabled) mail app, also looks way different

2.22:
![image](https://github.com/thp/urlwatch/assets/5646390/28a058d3-ae86-4884-b530-392321d3ad00)

2.28:
![image](https://github.com/thp/urlwatch/assets/5646390/2328c54a-eb95-4818-98ed-85c658f3640b)

Unfortunately this is extremely hard to read ...

Should this go to #730 ? 
What do you think @thp ?

**Comment by trevorshannon** [comment]
Dark mode in email clients is notoriously [hard](https://www.emailonacid.com/blog/article/email-development/dark-mode-for-email/) to nail.  I'll try to look into the differences between gmail and apple mail sometime. There must be a way to support both.  It kinda seems like iOS is double-inverting by inverting the css-specified dark mode background color of #121212 to something like #ededed

As for line heights, I think that the higher line height is definitely better in the HTML table view (as shown in #730), so I bet we can bring line height back to 1em just for plain text output.

**Comment by trevorshannon** [comment]
Ok, and #778 should fix the dark mode bug in Apple Mail on iOS.  Again, I could not test on an actual device, but rather used litmus.com to do so.

**Comment by kongomongo** [comment]
> As for line heights, I think that the higher line height is definitely better in the HTML table view (as shown in #730), so I bet we can bring line height back to 1em just for plain text output.

I'm sorry but what is HTML table view? Never heard of it.

**Comment by kongomongo** [comment]
> I made PR #777 to adjust the rendered line height for unified diff reports. Note I was not able to see the "spaced out" unified diff in Gmail--it must ignore line-height on `<body>` or something like that (I don't have Outlook). I did verify correct line height by rendering the HTML in Chrome, but feel free to test it out yourself using my branch

Works perfectly 👍



> Ok, and #778 should fix the dark mode bug in Apple Mail on iOS. Again, I could not test on an actual device, but rather used litmus.com to do so.

Also works! 👍 Looks like this on iOS:

**Comment by thp** [comment]
Thanks @trevorshannon for the quick and easy-to-review fixes, and @kongomongo for the report and verification :)

#777 and #778 have now been merged.
